build-monitor-plugin/src/main/java/com/smartcodeltd/jenkinsci/plugins/buildmonitor/viewmodel/BuildView.java

```diff
@@ -107,6 +115,21 @@ public String description() {
         return getOrElse(build.getDescription(), "");
     }
 
+    @Override
+    public List<String> stages() {
+        if (this.build instanceof WorkflowRun) {
+            WorkflowRun currentBuild = (WorkflowRun) this.build;
+
+            if (currentBuild.isBuilding()) {
+                StageTraversal traversal = new StageTraversal();
+                traversal.start(currentBuild.getExecution().getCurrentHeads());
```

Comment:
Update:

I tried this:
~~~
 if (currentBuild.isBuilding()) {
   StageTraversal traversal = new StageTraversal();
   if (currentBuild.getExecution() != null) {
     traversal.start(currentBuild.getExecution().getCurrentHeads());
     return traversal.getStages();
   }
}
~~~
and the NPEs seem to have gone away.